storage/src/test/java/org/apache/kafka/storage/internals/log/LogValidatorTest.java

```diff
@@ -0,0 +1,2228 @@
+                        CompressionType.NONE,
+                        Compression.NONE,
+                        false,
+                        RecordBatch.MAGIC_VALUE_V2,
+                        TimestampType.CREATE_TIME,
+                        1000L,
+                        1000L,
+                        RecordBatch.NO_PARTITION_LEADER_EPOCH,
+                        AppendOrigin.CLIENT,
+                        MetadataVersion.latestTesting()
+                ).validateMessagesAndAssignOffsets(
+                        PrimitiveRef.ofLong(0),
+                        metricsRecorder,
+                        RequestLocal.withThreadConfinedCaching().bufferSupplier()
+                )
+        );
+    }
+
+    @Test
+    public void testInvalidChecksum() {
+        checkInvalidChecksum(RecordBatch.MAGIC_VALUE_V0, Compression.gzip().build(), CompressionType.GZIP);
+        checkInvalidChecksum(RecordBatch.MAGIC_VALUE_V1, Compression.gzip().build(), CompressionType.GZIP);
+        checkInvalidChecksum(RecordBatch.MAGIC_VALUE_V0, Compression.lz4().build(), CompressionType.LZ4);
+        checkInvalidChecksum(RecordBatch.MAGIC_VALUE_V1, Compression.lz4().build(), CompressionType.LZ4);
+        checkInvalidChecksum(RecordBatch.MAGIC_VALUE_V0, Compression.snappy().build(), CompressionType.SNAPPY);
+        checkInvalidChecksum(RecordBatch.MAGIC_VALUE_V1, Compression.snappy().build(), CompressionType.SNAPPY);
+    }
+
+    private void checkInvalidChecksum(byte magic, Compression compression, CompressionType type) {
+        LegacyRecord record = LegacyRecord.create(magic, 0L, null, "hello".getBytes());
+        ByteBuffer buf = record.buffer();
+
+        // enforce modify crc to make checksum error
+        buf.put(LegacyRecord.CRC_OFFSET, (byte) 0);
+
+        ByteBuffer buffer = ByteBuffer.allocate(1024);
+        MemoryRecordsBuilder builder = MemoryRecords.builder(buffer, magic, compression,
+                TimestampType.CREATE_TIME, 0L);
+        builder.appendUncheckedWithOffset(0, record);
+
+        MemoryRecords memoryRecords = builder.build();
+        LogValidator logValidator = new LogValidator(memoryRecords,
+                topicPartition,
+                time,
+                type,
+                compression,
+                false,
+                magic,
+                TimestampType.CREATE_TIME,
+                1000L,
+                1000L,
+                RecordBatch.NO_PARTITION_LEADER_EPOCH,
+                AppendOrigin.CLIENT,
+                MetadataVersion.latestTesting()
+        );
+
+
+        assertThrows(CorruptRecordException.class, () -> logValidator.validateMessagesAndAssignOffsets(
+                PrimitiveRef.ofLong(0),
+                metricsRecorder,
+                RequestLocal.withThreadConfinedCaching().bufferSupplier()
+        ));
+
+        assertTrue(metricsRecorder.recordInvalidChecksumsCount > 0);
+    }
+
+    @Test
+    public void testInvalidSequence() {
```

Comment:
nice one. could you please use `EnumSource` to rewrite it? for example:
```java
    @ParameterizedTest
    @EnumSource(CompressionType.class)
    public void testInvalidSequenceV0(CompressionType type) {
        checkInvalidSequence(RecordBatch.MAGIC_VALUE_V0, type);
    }

    @ParameterizedTest
    @EnumSource(CompressionType.class)
    public void testInvalidSequenceV1(CompressionType type) {
        checkInvalidSequence(RecordBatch.MAGIC_VALUE_V2, type);
    }

    @ParameterizedTest
    @EnumSource(CompressionType.class)
    public void testInvalidSequenceV2(CompressionType type) {
        checkInvalidSequence(RecordBatch.MAGIC_VALUE_V2, type);
    }
```

Reply:
Hi @chia7712 , Thanks for review.
I rewrite this part, please take a look.